app.py

```diff
@@ -1,72 +1,64 @@
 #!/usr/bin/env python3
 
-import json
 import os
-import cherrypy
-from elasticsearch import Elasticsearch
-from lib.model.part import Part
-from lib.render import PHRender
+from pathlib import Path
 
+from elasticsearch import AsyncElasticsearch
+from quart import Quart, redirect, request, render_template, safe_join, send_file
 
-class Search:
-    def __init__(self):
-        self.phrender = PHRender()
+from lib.model.part import Part
 
-        self.env = self.phrender.env
-        self.template = self.phrender.get_class_template(self.__class__)
 
-        if 'ELASTICSEARCH' in os.environ:
-            self.es = Elasticsearch([os.environ['ELASTICSEARCH']])
-        else:
-            self.es = Elasticsearch()
+app = Quart(__name__)
 
-    def render(self, page=None):
-        """Render the template associated with the page.
+STATIC_DIR = Path(__file__).parent / '_site'
 
-        If the client requests JSON, then this returns a JSON dump of the
-        """
+@app.route("/", defaults={'path': ""})
+@app.route("/<path:path>")
+async def serve(path: str):
```

Comment:
It feels like there should be a static hosting utility that does this.

Reply:
I felt the same, and initially I used Quart's builtin static serving (see e7d1d22b661e1c8d933d7dd7e9b2383d3bce2c4a) but it didn't handle index files correctly (i.e. visiting `/` yielded a 404, instead of serving `/index.html`) and [according to Quart's dev] there's nothing builtin for that.

[according to Quart's dev]: https://gitlab.com/pgjones/quart/-/issues/425#note_623142004

Reply:
I felt the same, and initially I used Quart's builtin static serving (see e7d1d22) but it didn't handle index files correctly (i.e. visiting / yielded a 404, instead of serving /index.html) and [according to Quart's dev] there's nothing builtin for that.

[according to Quart's dev]: https://gitlab.com/pgjones/quart/-/issues/425#note_623142004